File: Snippets/Fibonacci/fib.rb
$fib            = {}
$fib_prev       = 0
$fib_last       = 1
$fib[$fib_prev] = true
$fib[$fib_last] = true

def is_fib (n)
    while $fib_last < n do
        t               = $fib_last
        $fib_last      += $fib_prev
        $fib_prev       = t
        $fib[$fib_last] = true
    end

    return $fib[n]
end

File: Snippets/Fibonacci/fib.go
var fib_prev, fib_last int  
var fib map [int] bool

func fib_init () {
    fib            = make (map [int] bool)
    fib_prev       = 0
    fib_last       = 1
    fib [fib_prev] = true
    fib [fib_last] = true
}


func is_fib (n int) bool {
    for fib_last < n {
        t             := fib_last
        fib_last      += fib_prev
        fib_prev       = t
        fib [fib_last] = true
    }

    _, ok := fib [n]

    return (ok)
}
